File: app/src/main/java/de/eloc/eloc_control_panel/helpers/DeviceInfo.java
package de.eloc.eloc_control_panel.helpers;

import android.annotation.SuppressLint;
import android.bluetooth.BluetoothDevice;

public class DeviceInfo {
    public static final String DEFAULT_NAME = "<unknown device>";
    public static final String DEFAULT_ADDRESS = "00:00:00:00:00:00";

    public final String name;
    public final String address;

    public DeviceInfo(String name, String address) {
        this.name = name;
        this.address = address;
    }

    public static DeviceInfo getDefault() {
        return new DeviceInfo(DEFAULT_NAME, DEFAULT_ADDRESS);
    }

    public static DeviceInfo fromDevice(BluetoothDevice device) {
        // it is safe to suppress permission,
        // because when this method is call, bluetooth permissions
        // will already be granted.
        if (device != null) {
            @SuppressLint("MissingPermission")
            String name = device.getName();
            if (name == null) {
                name = DeviceInfo.DEFAULT_NAME;
            }
            String address = device.getAddress();
            if (address == null) {
                address = DEFAULT_ADDRESS;
            }
            return new DeviceInfo(name, address);
        }
        return new DeviceInfo("<unknown device>", "<00:00:00:00:00:00>");

    }
}

File: app/src/main/java/de/eloc/eloc_control_panel/ng/models/DeviceInfo.kt
package de.eloc.eloc_control_panel.ng.models

import android.annotation.SuppressLint
import android.bluetooth.BluetoothDevice
import de.eloc.eloc_control_panel.ng2.models.BluetoothHelper

class DeviceInfo(val name: String, val address: String) {

    companion object {
        const val DEFAULT_NAME = "<unknown device>"
        private const val DEFAULT_ADDRESS = "00:00:00:00:00:00"

        val default: DeviceInfo = DeviceInfo(DEFAULT_NAME, DEFAULT_ADDRESS)

        @SuppressLint("MissingPermission")
        fun fromDevice(device: BluetoothDevice): DeviceInfo {
            if (!BluetoothHelper.instance.hasConnectPermission()) {
                return default
            }

            val name = device.name ?: DEFAULT_NAME
            val address = device.address ?: DEFAULT_ADDRESS
            return DeviceInfo(name, address)
        }
    }
}
